### PR TITLE
Remove unused dependencies

### PR DESCRIPTION
**Pull request by aoudiamoncef**
* [x] document all added functions
* [ ] try in sandbox /simulation/labnet
  * [ ] if part of node-launch, checked using the `resync_check` flag
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification

### CODE REVIEW

File: massa-consensus-exports/Cargo.toml [hunk 1]
```diff
@@ -8,17 +8,14 @@ edition = "2021"
...
 displaydoc = {workspace = true}
 nom = {workspace = true}
 serde = {workspace = true, "features" = ["derive"]}
 serde_json = {workspace = true}   # BOM UPGRADE     Revert to "1.0" if problem
 thiserror = {workspace = true}
-jsonrpsee = {workspace = true, "features" = ["server"]}
 tokio = {workspace = true, "features" = ["sync"]}
 mockall = {workspace = true, "optional" = true}   # BOM UPGRADE     Revert to {"version": "0.11.4", "optional": true} if problem
 massa_channel = {workspace = true}
-massa_hash = {workspace = true}
```
Comment (Leo-Besancon) [inline massa-consensus-exports/Cargo.toml:21]: I think there are check comments associated (see at the bottom of the diff page on github).
`massa_hash` is used in the doc comments examples (https://github.com/massalabs/massa/blob/main/massa-consensus-exports/src/export_active_block.rs#L144), so I think we can put it as [dev-dependencies] instead?

Can you check for similar behaviours?

**Review by Leo-Besancon — CHANGES_REQUESTED** [review]
We need to be sure the dependancies are not used with a feature combination / in comment examples before merging this one. Can you check?
Thanks for digging everything up :).

File: massa-consensus-exports/Cargo.toml (hunk 1 above)
Comment (aoudiamoncef) [reply massa-consensus-exports/Cargo.toml:21]: I'm agree, we have to check all combinations.

Comment (aoudiamoncef) [reply massa-consensus-exports/Cargo.toml:21]: It seems that the deps analyzer don't recognize the doctests for example

**Review by Leo-Besancon — APPROVED** [review]
Nice job :)

**Comment by aoudiamoncef** [comment]
bors merge

**Pull request merged**